### ISSUE TITLE
[DataSource Region][Feature request] Provide the current region set by env SCALINGO_REGION

### ISSUE THREAD

**Issue by josephpage** [Scalingo/terraform-provider-scalingo#152]
When I use the provider with the region set by the SCALINGO_REGION environment variable, I haven't found an easy way to get the region to use elsewhere (for an `output` for example). The reason is that unlike the TF providers of other cloud services ([`aws` for example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/region#example-usage)), the Data Source `scalingo_region` requires to provide a region name and does not return the "current" region by default.

Ideally, I would like the following code to work:

```tf
required_providers {
    scalingo = {
        source  = "Scalingo/scalingo"
        version = "~> 2.0"
    }
}
    
provider "scalingo" {
    # SCALINGO_API_TOKEN and SCALINGO_REGION set by env vars
}

data "scalingo_region" "current" {}

# displays the region declared in SCALINGO_REGION env var
output "current_scalingo_region" {
    value = data.scalingo_region.current 
}
```

In case the region is not configured in the `provider` block or in the environment variable, you have to return an error and block the plan/apply.

**Comment by guillaume-sc** [comment]
Thank you for the suggestion, @josephpage. 

Would defining a Terraform variable work in your case? For example, in the environment, `export TF_VAR_scalingo_region=osc-fr1`. Then, in Terraform file `main.tf`, define the following:
```terraform
var "scalingo_region" {}

data "scalingo_region" "current" {
  name = var.scalingo_region
}
```

**Comment by josephpage** [comment]
Thank you for your response @guillaume-sc. The workaround you suggest works, of course, but it requires declaring the region in 2 different environment variables (`SCALINGO_REGION` for the provider and `TF_VAR_scalingo_region ` for the region) which is not ideal (and not really my definition of simple).

Because I like the simplicity and the explicit name of `SCALINGO_REGION`, I implemented a workaround that works with only one environment variable, but it is also quite verbose. :

```
required_providers {
    environment = {
        source  = "EppO/environment"
    }
}

data "environment_variables" "scalingo_region_env" {
  filter = "^SCALINGO_REGION"
}

data "scalingo_region" "current" {
  name = data.environment_variables.scalingo_region_env.items[SCALINGO_REGION]
}
```

I have encapsulated this solution in the open-source module that I maintain : https://github.com/scalingo-community/terraform-scalingo-app/blob/main/output.tf#L18
Nevertheless, I think that the Scalingo provider should offer this feature natively.

My practical use case is to retrieve the deployment region of an application to display it as metadata for the StatusCake alert that regularly checks the health of the application : https://github.com/DNUM-SocialGouv/1j1s-front/blob/terraform/terraform/healthcheck.tf#L30
